app/models/notification.rb

```diff
@@ -0,0 +1,20 @@
+class Notification < ApplicationRecord
+  belongs_to :notification_type
+  belongs_to :initiator, :class_name => User, :foreign_key => 'user_id'
+  belongs_to :subject, :polymorphic => true
+  belongs_to :cause, :polymorphic => true
+  has_many :notification_recipients, :dependent => :delete_all
+  has_many :recipients, :class_name => User, :through => :notification_recipients
+
+  def self.emit(type, opts = {})
```

Comment:
This can be moved into `initialize` with [callbacks](http://guides.rubyonrails.org/active_record_callbacks.html).